File: AquaTrack/src/main/java/com/aquatrack/service/ResidentInvitationService.java
package com.aquatrack.service;

import com.aquatrack.dto.resident.CreateResidentInvitationRequestDto;
import com.aquatrack.dto.resident.ResidentActivationDetailsResponseDto;
import com.aquatrack.dto.resident.ResidentActivationRequestDto;
import com.aquatrack.dto.resident.ResidentInvitationResponseDto;

import java.util.List;

public interface ResidentInvitationService {

    // ==========================================
    // Create Resident Invitation
    // ==========================================

    ResidentInvitationResponseDto createResidentInvitation(
            CreateResidentInvitationRequestDto requestDto
    );

    // ==========================================
    // Get All Resident Invitations
    // ==========================================

    List<ResidentInvitationResponseDto>
    getAllResidentInvitations();

    // ==========================================
    // Get Resident Invitation By ID
    // ==========================================

    ResidentInvitationResponseDto
    getResidentInvitationById(
            Long invitationId
    );

    // ==========================================
    // Get Activation Details
    // ==========================================

    ResidentActivationDetailsResponseDto
    getActivationDetails(
            String token
    );

    // ==========================================
    // Activate Resident
    // ==========================================

    void activateResident(
            ResidentActivationRequestDto requestDto
    );

}

File: Aquatrack-frontend/src/services/residentInvitationService.js
import api from "../api/axios";

const residentInvitationService = {

    // ==========================================
    // Create Resident Invitation
    // ==========================================

    async createResidentInvitation(data) {

        const response = await api.post(
            "/resident-invitations",
            data
        );

        return response.data;

    },

    // ==========================================
    // Get All Invitations
    // ==========================================

    async getAllResidentInvitations() {

        const response = await api.get(
            "/resident-invitations"
        );

        return response.data;

    },

    // ==========================================
    // Get Invitation By ID
    // ==========================================

    async getResidentInvitationById(invitationId) {

        const response = await api.get(
            `/resident-invitations/${invitationId}`
        );

        return response.data;

    },

    // ==========================================
    // Get Activation Details
    // ==========================================

    async getActivationDetails(token) {

        const response = await api.get(
            `/resident-invitations/activate?token=${token}`
        );

        return response.data;

    },

    // ==========================================
    // Activate Resident
    // =========================================

    async activateResident(data) {

        const response = await api.post(
            "/resident-invitations/activate",
            data
        );

        return response.data;

    }

};

export default residentInvitationService;